pkg/util/validation/validate.go

```diff
@@ -65,12 +65,14 @@ func (cfg *Overrides) ValidateSample(userID string, metricName []byte, s client.
 // ValidateLabels returns an err if the labels are invalid.
 func (cfg *Overrides) ValidateLabels(userID string, ls []client.LabelPair) error {
 	metricName, err := extract.MetricNameFromLabelPairs(ls)
-	if err != nil {
-		return httpgrpc.Errorf(http.StatusBadRequest, errMissingMetricName)
-	}
+	if cfg.EnforceMetricName(userID) {
```

Comment:
In the case where we don't enforce, and it is blank, the error returns below (lines 81 and 105) become less usable.

Reply:
Yeah, bit of a shame.  Could return the whole label set there - WDYT?